CONTRIBUTING.rst

```diff
@@ -0,0 +1,480 @@
+
+If you are writing and debugging a test, you may find it helpful to have
+print statements printed to the terminal while running tests (which
+doesn't happen by default). To do this,
+
+::
+
+   >> pytest -s
+
+If you want to only run a specific test function within a file (or only
+tests with names that contain a certain string):
+
+::
+
+   >> pytest -k testname.
+
+This will search all files under the directory you're in for files or
+functions with the string 'test' in them, and within those files run
+only the test functions that contain the string ``testname``.
+
+Within the test files themselves, decorators can be used to control the
+behavior of the test. Some of the more useful decorators include: 1.
+@pytest.mark.parametrize can be used to run a single test on multiples
+sets of input parameters 2. @pytest.skip can be used to skip tests
+altogether, or under specific conditions (for example, only when being
+run by the CI) 3. @pytest.fixture to declare a
+`fixture <https://docs.pytest.org/en/7.1.x/explanation/fixtures.html>`__
+1. @pytest.mark.xfail will make a test pass only if it fails.
+
+Simultaneously developing ``drizzlepac`` and one of its dependencies
+--------------------------------------------------------------------
+
+If you encounter the scenario where you wish to simultaneously make
+changes in ``drizzlepac`` and also in one of its dependencies like
+``stcal`` or ``stpipe``, we recommend that you create a new environment
+with development versions of both of these packages. To do this, you can
+follow the same workflow outlined in the 'Contributing code' section of
+this guide. To summarize, you will want to create a new Python
+environment (called, for example, drizzlepac_stcal_dev), fork and clone
+a local copy of ``drizzlepac`` if you haven't already and install this
+(doing ``pip install -e .`` in the top level directory of
+``drizzlepac``), and then fork and clone ``stcal`` and install it in
+this environment in the same way. To double check that you have the
+correct dev versions of these packages in your environment, you can
+check their versions by doing:
+
+::
+
+   >> conda list
+
+When opening up two dependent pull requests in ``drizzlepac`` and one of
+its dependency packages, unit tests will not pass on the CI because the
+``pyproject.toml`` file in ``drizzlepac`` points to the last released
+version of ``stcal``, and stcal points to the last version of
+``drizzlepac``, so the issue becomes circular. What you will need to do
+is modify the ``pyproject.toml`` files in both packages to point to the
+other to demonstrate that CI tests pass (and make a comment noting this
+in your PR), and then change it back before the PR is merge so that
+changes to ``pyproject.toml`` are not merged into main. In your
+``drizzlepac`` branch, to point to your branch in the dependent package
+(in this example ``stcal``), change the required ``stcal`` version in
```

Comment:
Change references to 'stcal' to a package which is really a dependency of 'drizzlepac'.